### ISSUE TITLE
Suggestion: Arbitrary block to arbitrary block custom plants

### ISSUE THREAD

**Issue by GavJenks** [DevotedMC/RealisticBiomes#10]
For example, a violet tulip which matures into a lilac plant which I might call "flax" and then make drop wool. Or a vine matures into a purple wool "grape" or a azure bluet matures into a peony which i've reskinned into cotton.

I think the way RB works (?) is to have a collection of plant locations loaded and running a timed pass over them now and then, then comparing timestamps on chunk loading. If so, there's no need for the plant to be an actual minecraft plant, is there? It might not be very much extra work to allow configurable custom from:to plant types that could triple or quadruple the gameplay latitude of the plugin almost overnight.

**Comment by ProgrammerDan** [comment]
See, this is awesome. But definitely more for CropControl -- this is an explicit desire for CropControl in terms of expansion / improvements.

**Comment by ProgrammerDan** [comment]
Let me expand on that. Realistic Biomes isn't about maturation _cycles_ but rather maturation time. CropControl is explicitly about maturation cycles AND drop control, which is explicitly what you are describing. Instead of bloating Realistic Biomes, we made CropControl to do exactly what you are describing, but initial implementation just wraps currently available crops; future implementations should include exactly what you are describing (custom growth patterns w/ custom drops, etc).

**Comment by ProgrammerDan** [comment]
> I think the way RB works (?) is to have a collection of plant locations loaded and running a timed pass over them now and then, then comparing timestamps on chunk loading.

Not exactly, should have mentioned that. It does do timestand comparing and blending to control / prevent growth cycles, but only for existing plants that trigger their own maturities. And even then, quite limited. Def something more suited for CropControl, which already has deep growth cycle settings and it would be a pretty natural extension to allow for custom growth types, natures, and drops.
